application/models/artifact.py

```diff
@@ -37,6 +40,30 @@ def tags_list(self):
             tags_list.append(tag.name)
         return tags_list
 
+    @property
+    def user_tags_list(self):
+        """ Returns this artifacts user entered tags as string array """
+        tags_list = []
+        for tag in self.user_tags:  # pylint:disable=not-an-iterable
+            tags_list.append(tag.name)
+        return tags_list
+
+    @property
+    def label_tags_list(self):
+        """ Returns this artifacts image recognition generated tags as string array """
+        tags_list = []
+        for tag in self.label_tags:  # pylint:disable=not-an-iterable
+            tags_list.append(tag.name)
+        return tags_list
+
+    @property
+    def text_tags_list(self):
+        """ Returns this artifacts text recognition generated tags as string array """
+        tags_list = []
+        for tag in self.text_tags:  # pylint:disable=not-an-iterable
+            tags_list.append(tag.name)
+        return tags_list
```

Comment:
this is all but pretty now, we either need to be able to create these getters dynamically. Or find a different way to make the iterators accessible for the schema.

One thing that definitly makes this all a bit better already though would simply be `return list(self.tags)` but even that is too much duplication imo.